File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/typeconverter/JSONConverter.java
package com.raizlabs.android.dbflow.test.typeconverter;

import com.raizlabs.android.dbflow.converter.TypeConverter;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Description:
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
public class JSONConverter extends TypeConverter<String, JSONObject> {
    @Override
    public String getDBValue(JSONObject model) {
        return model.toString();
    }

    @Override
    public JSONObject getModelValue(String data) {
        JSONObject jsonObject = null;
        try {
            jsonObject = new JSONObject(data);
        } catch (JSONException e) {

        } finally {
            return jsonObject;
        }
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/typeconverter/JSONConverter.kt
package com.raizlabs.android.dbflow.test.typeconverter

import com.raizlabs.android.dbflow.converter.TypeConverter

import org.json.JSONException
import org.json.JSONObject

/**
 * Description:
 */
@com.raizlabs.android.dbflow.annotation.TypeConverter
class JSONConverter : TypeConverter<String, JSONObject>() {

    override fun getDBValue(model: JSONObject?): String? {
        return model?.toString()
    }

    override fun getModelValue(data: String): JSONObject? {
        var jsonObject: JSONObject? = null
        try {
            jsonObject = JSONObject(data)
        } catch (e: JSONException) {

        } finally {
            return jsonObject
        }
    }
}
